pkg/currency/module.go

```diff
@@ -11,10 +12,12 @@ import (
 )
 
 type Module struct {
-	kernel.BackgroundModule
+	kernel.HealthCheckedModule
```

Comment:
I don't think you want to embed the interface here. If you want to ensure the module implements the interface, use `var _ kernel.HealthCheckedModule = &Module{}`